src/engine/runtime.js

```diff
@@ -1522,7 +1522,18 @@ class Runtime extends EventEmitter {
      * @param {!MonitorRecord} monitor Monitor to add.
      */
     requestAddMonitor (monitor) {
-        this._monitorState = this._monitorState.set(monitor.get('id'), monitor);
+        const id = monitor.get('id');
+        if (this._monitorState.has(id)) {
```

Comment:
Maybe you can just use `requestUpdateMonitor` here instead of defining the functionality twice. `requestUpdateMonitor` can return a boolean representing whether a monitor already existed in the state, and then if it didn't, you can set it here as in the else case below.